src/Cardano/Address.hs

```diff
@@ -130,6 +134,19 @@ class PaymentAddress key
             -- ^ Staking key
         -> Address
 
+    -- | Derive a staking key for a corresponding 'AccountK'. Note that wallet
+    -- software are by convention only using one staking key per account, and always
+    -- the first account (with index 0').
+    --
+    -- Deriving staking keys for something else than the initial account is not
+    -- recommended and can lead to incompatibility with existing wallet softwares
+    -- (Daedalus, Yoroi, Adalite...).
+    --
+    -- @since 2.0.0
+    deriveStakingPrivateKey
+        :: key 'AccountK XPrv
+        -> key 'StakingK XPrv
+
```

Comment:
I don't think this is the right place to have this one. I'd suggest creating another type-class to capture this constraint as

```hs
class StakingDerivation key where
   ...
```

Very much like we have `SoftDerivation` and `HardDerivation`.

Reply:
done